keps/sig-node/4188-kubelet-pod-readiness-api/README.md

```diff
@@ -217,75 +240,90 @@ the Pods. We don't want this API to return partial data.
+freshest possible information about pods running on the node.
+
 ### Proposed API
 
-We propose to add new gPRC API `status` in Kubelet, listening on a unix socket
-at `/var/lib/Kubelet/status/Kubelet.sock`. The endpoint will be versioned. The
-gRPC Service will expose 3 methods serving local Pods statuses data:
+We propose to add new gRPC API `pods` in Kubelet, listening on a unix socket
+at `/var/lib/Kubelet/pods/Kubelet.sock`. The endpoint will not be versioned. The
+gRPC Service will expose 3 methods serving local Pods data:
 
 ```protobuf
-service PodStatus {
-    // ListPodStatus returns a of List of PodStatus
-    rpc ListPodStatus(PodStatusListRequest) returns (PodStatusListResponse) {}
-    // WatchPodStatus returns a stream of List of PodStatus
-    // Whenever a pod state change api returns the new list
-    rpc WatchPodStatus(PodStatusWatchRequest) returns (stream PodStatusWatchResponse) {}
-    // GetPodStatus returns a PodStatus for given pod's UID
-    rpc GetPodStatus(PodStatusGetRequest) returns (PodStatusGetResponse) {}
+import "google/protobuf/field_mask.proto";
+
+service Pods {
+    // ListPods returns a list of v1.Pod, filtered by field mask.
+    rpc ListPodStatus(PodListRequest) returns (PodListResponse) {}
+    // WatchPods returns a stream of list of PodInfo, filtered by field mask.
+    // Whenever a pod state changes, api returns the new list.
+    rpc WatchPods(PodWatchRequest) returns (stream PodWatchResponse) {}
+    // GetPod returns a PodInfo for given pod's UID, filtered by field mask.
+    rpc GetPod(PodGetRequest) returns (PodGetResponse) {}
 }
 
-// PodCondition aligns with v1.PodCondition.
-message PodCondition {
-    PodConditionType Type = 1;
-    ConditionStatus Status = 2;
-    Timestamp LastProbeTime = 3;
-    Timestamp LastTransitionTime = 4;
-    string Reason = 5;
-    string Message = 6;
+// PodListRequest allows specifying a field mask.
+message PodListRequest {
+  // Optional field mask in the gRPC metadata.
 }
 
-// PodConditionType aligns with v1.PodConditionType
-enum PodConditionType {
-    ContainersReady = 0;
-    Initialized = 1;
-    Ready = 2;
-    PodScheduled = 3;
-    DisruptionTarget = 4;
+// PodListResponse returns a list of v1.Pod.
+message PodListResponse {
+  repeated v1.Pod pods = 1;
 }
 
-// ConditionStatus aligns with v1.ConditionStatus
-enum ConditionStatus {
-    True = 0;
-    False = 1;
-    Unknown = 2;
+// PodWatchRequest allows specifying a field mask.
+message PodWatchRequest {
+  // Optional field mask in the gRPC metadata.
 }
 
-// PodStatus returns a Pod details and list of status Conditions with deletion info.
-message PodStatus {
-    string podUID = 1;
-    string podNamespace = 2;
-    string podName = 3;
-    bool static = 4;
-    repeated PodCondition conditions = 5;
-    Timestamp DeletionTimestamp = 3;
+// PodWatchResponse returns a v1.Pod, as a stream.
```

Comment:
should this return the whole Pods stream or just Pod GUID? It may be cheaper for client to filter for Pods it wants to know about

Reply:
this is where the fieldmask is valuable, @briansonnenberg can you chime in?